### PR TITLE
feat: new snowflake data extraction pipeline for nimbus okr and platform optimizations

### PR DESCRIPTION
**Pull request by braunreyes**
## Goal
What changed? What is the business/product goal?

This PR contains the initial pipeline for extracting snowplow events to GCS storage (via snowflake external stage).
As part of this change, we have a bunch of optimizations to the platform, which make the pipeline possible and make things even easier for folks.  Below is an index of the big changes made:

- `.circleci/workflows.yml`: 
  - added the `DF_CONFIG_DEPLOYMENT_TYPE` envar to the ECS execution environment for CommonSettings class to work.
- `common-utils/pyproject.toml`: 
  - moved pytest config here.
- `common-utils/pytest.ini`: 
  - removed in favor of using pyproject.toml
- `common-utils/src/common/__init__.py`: 
 - helper for getting the script path for flow file.
- `common-utils/src/common/cloud/gcp_utils.py`: 
  - adjusting the `data` kwargs is best practice for Pydantic versus updating via `super()`
- `common-utils/src/common/databases/snowflake_utils.py`:
  - need role and private key arguments for properly creating the snowflake credentials block
  - enforcing the warehouse used must be a `prefect` warehouse.  We can always added shared use warehouse naming patterns later on if needed.
  - enforcing the database is either `development` or `prefect` on the connection.
   - new helper for getting the GCS Snowflake external stage for Prefect based on environment.
   - adjusting the `data` kwargs is best practice for Pydantic versus updating via `super()`
   - allow for setting the schema and warehouse on the connection via settings or explicitly on the connector class
- `common-utils/src/common/settings.py`:
  - new helper properties for handing deployment types as high level dev or prod environments
- `common-utils/tests/conftest.py`:
  - making the prefect test harness global
- `data-products/src/orchestration/main_orchestration_flow.py`:
  - killing the initial example flow used for testing
-  `data-products/src/snowflake_query_extraction/snowflake_query_extraction_flow.py` - new query extraction flow
## Deployment steps
- [X] Database migrations?
- [X] Secrets?
- [N/A] **FOR POCKET DEVELOPERS ONLY** - Post in [\#changelog](https://pocket.slack.com/archives/C0Q4UFMDZ):
> Write a #changelog message using our [best practices](https://docs.google.com/document/d/1oEt8Mtkp-6Xz9S2zaNX1EIXfQIEDN2JpY0diuPc1HZc/edit) if this PR (potentially) impacts users. Describe the 'why' and 'what'. @mention relevant teams. Link to this PR.

## References

JIRA ticket:
https://getpocket.atlassian.net/browse/DIS-664

Documentation:
https://getpocket.atlassian.net/wiki/spaces/PE/pages/2839904307/Prefect+v2+Migration
Will be adding project docs static site via [DIS-808](https://getpocket.atlassian.net/browse/DIS-808)

[DIS-808]: https://getpocket.atlassian.net/browse/DIS-808?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by braunreyes** [comment]
had to make some changes to how secrets are referenced...testing dev shows its working as expected:

image

**Comment by braunreyes** [comment]
final testing with last set of commits
image

**Review by chelseybeck — APPROVED** [review]
this looks good to me @braunreyes...I appreciate the descriptive comments and extensive testing

File: common-utils/src/common/settings.py [hunk 1]
```diff
@@ -35,3 +35,27 @@ class CommonSettings(Settings):
...
+        """Shortcut to get an explicit
+        dev' or 'production'.
+
+        Returns:
+            str: 'dev' or 'production'
+        """
+        answer = "dev"
+        if self.is_production:
+            answer = "production"
+        return answer
```
Comment (jeshuaborges) [inline common-utils/src/common/settings.py:61]: supper nitty, but why require the use of an `answer` variable in this function?

File: data-products/src/snowflake_query_extraction/snowflake_query_extraction_flow.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    )
+    logger.info("Running extraction...")
+    # run the extraction
+    extract = await snowflake_query.with_options(name="run-extraction")(  # type: ignore
+        query=extraction_job.extraction_sql,
+        snowflake_connector=sfc,
+    )
+    logger.info("Applying new state...")
+    # commit the new offset
+    await snowflake_query.with_options(name="persist-new-offset")(  # type: ignore
```
Comment (jeshuaborges) [inline data-products/src/snowflake_query_extraction/snowflake_query_extraction_flow.py:197]: I dont remember do we need to return anything to confirm success or is it assumed if there are no exceptions?

**Review by jeshuaborges — COMMENTED** [review]
Everything looks super good here. I threw in some nits and questions.

At a high level I do find myself wiching the code were more or documentation was specific about how the flow was moving data between these systems. You have to get pretty deep to find it. How is it staged and then loaded....

**Pull request merged**